src/readData.c

```diff
@@ -0,0 +1,48 @@
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+char *readData(char *file)
+{
+	int fd=-1;
+	char *buff = NULL;
+	char *dup = NULL;
+	int buffLength;
+	int i;
+	char tmp = 0;
+	int reat;
+
+	fd = open(file, O_RDONLY);
+	if (fd == -1)
+		goto F_READDATA_END;
```

Comment:
```suggestion
		return NULL;
```
Rien n'est alloué, tu peux juste `return NULL`